main/resource-access.bs

```diff
@@ -130,6 +130,86 @@ Note:
 Servers that wish to disable URI re-use may want to use the `410` status
 code.
 
+### Resource type heuristics ### {#resource-type-heuristics}
+
+Servers MUST apply the following heuristics to determine a request's resource
+type
+[[Source](https://github.com/solid/specification/issues/128#issuecomment-573033297)].
+
+<table id="heuristics-resource-type">
+  <caption>Heuristics to determine a resource type</caption>
+  <thead>
+    <tr>
+      <th>`Slug`</th>
+      <th>`Link`</th>
+      <th>Result</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <td>`foo/`</td>
+      <td>No `Link` header.</td>
+      <td>Create a container with `foo/` appended to the request-URI.</td>
+    </tr>
+    <tr>
+      <td><code>foo/</code></td>
+      <td>`Link` header with `ldp:Container` or a specialisation.</td>
+      <td>Create a container with `foo/` appended to the request-URI.</td>
+    </tr>
+    <tr>
+      <td><code>foo/</code></td>
+      <td>`Link` header other than container type.</td>
+      <td>Create a resource with `foo` appended to the request-URI, check for
```

Comment:
This exception I find very weird. I don't think we should derive anything from non-recognized `Link` headers. I.e., all `no` and `other` cases should be identical for me.

Reply:
That's indeed what's intended. What's derived is that it is a non-Container resource. Slug will be ignored in this case. Would rephrasing help?

Reply:
My interpretation of what written seems to be correct (so writing is ok); but I don't understand why from an unknown header can be derived the resource is non-Container. I don't think we should derive that.

Reply:
What's recognised is "ldp:Container or a specialisation" (ldp:BasicContainer - as that's the only one currently required) as the "container type". I used "other than container type" to express their absence. Open to rephrasing.

Reply:
Ah okay, so only `Link` header values with a `type` link relation are included then? Or do we really mean _all_ `Link` header values?

Reply:
Only the existence of Link with rel=type LDPC or LDP-BC is checked.

Note the check in prior row says "Link header with ldp:Container or a specialisation." - We are really talking about rel=type here.

So we can just mention the actual relation.

Reply:
Yes, I would make it clear that it is a `rel=type` header value. Then I have no further objections.

We could perhaps call them "indicated types" or so, and then declare above the table how those indicated types are extracted from `Link`. No strong opinion.

Reply:
I used "or specialisation" instead of ldp:BasicContainer figuring that if other container types are supported in the future, they will also follow the slash semantics. If server gets Link rel=type LDP-DC, it will create a container resource with trailing slash.

Reply:
All fine with me, my issue/misunderstanding was in the phrasing of "`Link` header other than container type."

Reply:
How about https://github.com/solid/specification/pull/160/commits/e40d685fcb1ed70484ba3d1e093eecfa5563a0d3 for this draft?